test/inductor/test_torchinductor.py

```diff
@@ -8691,6 +8691,95 @@ def fn(input_ids) -> torch.Tensor:
         x = torch.randn(2, 2)
         self.common(fn, (x,), atol=0, rtol=0)
 
+    @staticmethod
+    def _check_resize_common(self, fn, x, size_or_y, memory_format, inplace):
+        x_ref_arg = x.clone()
+        x_opt_arg = x.clone()
+        x_numel = x.numel()
+        torch._dynamo.reset_code_caches()
+        opt_fn = torch._dynamo.optimize_assert(compile_fx)(fn)
+        correct = fn(x_ref_arg, size_or_y, memory_format)
+        actual = opt_fn(x_opt_arg, size_or_y, memory_format)
+
+        def get_numel(size_or_y):
+            if isinstance(size_or_y, torch.Tensor):
+                return size_or_y.numel()
+            else:
+                # assume shape
+                return functools.reduce(lambda x, y: x * y, size_or_y, 1)
+
+        nele_check = min(x_numel, get_numel(size_or_y))
+        correct_values = correct.as_strided((nele_check,), (1,))
+        # sanity check
+        self.assertTrue(same(x.as_strided((nele_check,), (1,)), correct_values))
+        actual_values = actual.as_strided((nele_check,), (1,))
+        self.assertTrue(same(correct_values, actual_values))
+        correct_strides = correct.stride()
+        actual_strides = actual.stride()
+        self.assertEqual(correct_strides, actual_strides)
```

Comment:
@lezcano I never replied to [this comment](https://github.com/pytorch/pytorch/pull/122317#discussion_r1543430877) but this assert will fail if we don't work to make sure the output has the correct stride order. 

Specifically, cases where `memory_format` is either  `torch.channels_last` or  `torch.channels_last_3d` for both ops. In addition cases where `memory_format` is `torch.preserve_format` and `size_or_y.suggest_memory_format()` is `torch.channels_last` or `torch.channels_last_3d` for `resize_as` (in these cases `size_or_y` will be a Tensor).